packages/aws-cdk/lib/api/util/sdk.ts

```diff
@@ -290,4 +331,62 @@ function httpsProxyFromEnvironment(): string | undefined {
         return process.env.HTTPS_PROXY;
     }
     return undefined;
-}
\ No newline at end of file
+}
+
+/**
+ * Return whether it looks like we'll have ECS credentials available
+ */
+function hasEcsCredentials() {
+    return (AWS.ECSCredentials.prototype as any).isConfiguredForEcsCredentials();
```

Comment:
Sweet, some more esoteric code of unholy nature doing something that the JS SDK should expose more directly \o/